### PR TITLE
yara: 4.3.1 -> 4.3.2

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/VirusTotal/yara/compare/v4.3.1...v4.3.2

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/yara/default.nix [hunk 1]
```diff
@@ -15,13 +15,13 @@
 
 stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/tools/security/yara/default.nix:16]: Here, I recommend using the `finalAttrs` pattern. To learn more about it, I suggest referring to the following resources:

1. Historical Discussion: The discourse thread at https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293 provides insights into the benefits of avoiding recursive expressions in Nixpkgs.
2. Documentation: The Nixpkgs manual provides detailed information on `mkDerivation` and recursive attributes. You can find it at https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes.
3. Implementation Example: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe.

I hope these resources provide valuable insights and guidance for your needs. If you have any further questions, feel free to ask.

**Review by drupol — COMMENTED** [review]
Hi,

Thanks for your PR!

I added a non-blocking comment, it would be super nice if you could implement it.
Regardless of your decision, I'll merge this PR at some point anyway.

Thanks in advance!

File: pkgs/tools/security/yara/default.nix [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
   pname = "yara";
-  version = "4.3.1";
+  version = "4.3.2";
 
   src = fetchFromGitHub {
     owner = "VirusTotal";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-Q+Q52W/MhurG3x0CIr0nv31qc4bdaLDk9AGGpMxKOcI=";
+    repo = finalAttrs.pname;
```
Comment (drupol) [inline pkgs/tools/security/yara/default.nix:19]:
```suggestion
    repo = "yara";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/yara/default.nix [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-TmMHn/Zi/ZyJGHobEg9r1uLWu7e+8XJPwCvnUhQlD6I=";
   };
 
+  withCrypto = true ;
```
Comment (kirillrdy) [inline pkgs/tools/security/yara/default.nix:24]: apologies for the question

is this change because its preferred to use `overrideAttrs` over `override` ?

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 237503` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.karton-autoit-ripper</li>
    <li>python310Packages.karton-config-extractor</li>
    <li>python310Packages.malduck</li>
    <li>python311Packages.karton-autoit-ripper</li>
    <li>python311Packages.karton-config-extractor</li>
    <li>python311Packages.malduck</li>
  </ul>
</details>
<details>
  <summary>37 packages built:</summary>
  <ul>
    <li>aaxtomp3</li>
    <li>acquire (python310Packages.acquire)</li>
    <li>apkid</li>
    <li>arch-install-scripts</li>
    <li>bash-preexec</li>
    <li>bashup-events32</li>
    <li>bashup-events44</li>
    <li>bats</li>
    <li>dgoss</li>
    <li>git-ftp</li>
    <li>imhex</li>
    <li>ix</li>
    <li>locate-dominating-file</li>
    <li>mons</li>
    <li>msmtp</li>
    <li>packcc</li>
    <li>pdf2odt</li>
    <li>pdfmm</li>
    <li>python310Packages.dissect</li>
    <li>python310Packages.dissect-target</li>
    <li>python310Packages.karton-yaramatcher</li>
    <li>python310Packages.yara-python</li>
    <li>python311Packages.acquire</li>
    <li>python311Packages.dissect</li>
    <li>python311Packages.dissect-target</li>
    <li>python311Packages.karton-yaramatcher</li>
    <li>python311Packages.yara-python</li>
    <li>s0ix-selftest-tool</li>
    <li>shunit2</li>
    <li>spyre</li>
    <li>stacs</li>
    <li>volatility3</li>
    <li>wgnord</li>
    <li>yadm</li>
    <li>yara</li>
    <li>yaralyzer</li>
    <li>zxfer</li>
  </ul>
</details>

File: pkgs/tools/security/yara/default.nix (hunk 3 above)
Comment (drupol) [reply pkgs/tools/security/yara/default.nix]: I'm also wondering the same question. I have the feeling that this is not what it is supposed to be done.

Comment (drupol) [reply pkgs/tools/security/yara/default.nix]: @fabaff ^^

**Comment by Stunkymonkey** [comment]
@fabaff can we close this due to the update to `4.4.0` in #261499 ?

**Pull request closed without merging**